README.md

```diff
@@ -92,3 +92,24 @@ the same directory (either top level or under a series of subdirectories called
 archive. On Windows, DMLC is sometimes unable to resolve these symlinks
 correctly; for this reason, it is recommended that the archive is only
 extracted and compiled on Linux.
+
+### DMLC_GATHER_SIZE_STATISTICS
+When set, DMLC outputs a file ending with `-size-stats.json`, which shows code
+generation statistics useful to reduce generated code size and increase compile
+speed. The file lists how much C code is generated for each DML method. The
+output is a list of triples `[tot_size, location, num]`, where `tot_size` is
+the total number of bytes of C code generated from one method declarations,
+`num` is how many times C code was generated from this declaration (because it
+was expanded by a template), and `location` is the source location of the
+declaration.
+
+An entry with a large `tot_size` and a large `num` can be reduced by declaring
+the method as `shared`; this should roughly divide the size by `num`. An entry
+with large `tot_size` with `num` equals 1 usually means the code can be reduced
+either by changing a `#foreach` or `#select` statement into `foreach` or
+`select`, or by breaking out the loop body to a separate method.
```

Comment:
> with large `tot_size` but where `num` is small &mdash; 1 to 3 &mdash; usually means
Also your suggestions here don't really apply. For one, we don't have a non-hash `select` statement, and for another, it's rather rare that `foreach` can actually be used to replace `#foreach`! I'd say "by breaking out the loop body to a separate body, or devising an alternate way of performing the loop such that it doesn't need to make use of `#foreach`/`#select`".

Reply:
Fixed.